### PR TITLE
Feat/migrate frontpage to drizzle

### PR DESCRIPTION
**Pull request by JohnAllenTech**
# ✨ Codu Pull Request 💻

Codu Logo

## Pull Request details

- Switched over the frontpage from using Prisma to Drizzle
- Fixed a bug where bookmarked posts were not appearing as bookmarked on the frontpage

## Any Breaking changes

- None

## Associated Screenshots

image

### CODE REVIEW

File: server/lib/posts.ts [hunk 1]
```diff
@@ -73,40 +70,45 @@ export async function getTrending({ currentUserId }: GetTrending) {
...
 
-    const cleaned = response.map((post) => {
-      let currentUserLikesPost = !!post.bookmarks.length;
-      if (currentUserId === undefined) currentUserLikesPost = false;
-      post.bookmarks = [];
-      return { ...post, currentUserLikesPost };
+    const cleaned = response.map((elem) => {
+      let currentUserBookmarkedPost = !!elem.bookmarked;
+      if (currentUserId === undefined) currentUserBookmarkedPost = false;
+      return { ...elem.post, user: elem.user, currentUserBookmarkedPost };
```
[the remark this answers is not in the document]
Comment (JohnAllenTech) [reply server/lib/posts.ts]: help me if you can

[the remark this answers is not in the document]
Comment (JohnAllenTech) [reply server/lib/posts.ts]: I am going to try this out and get back to you bot. If this works ill be highly impressed

[the remark this answers is not in the document]
Comment (JohnAllenTech) [reply server/lib/posts.ts]: Humans rule. Bots drool. Your code doesnt work as drizzle doesnt accept undefined params in its select options

bookmarked: bookmarked ? { id: bookmarked.id } : undefined

is showing as a ts error. It probably works in js though 🤷🏻‍♂️

[the remark this answers is not in the document]
Comment (JohnAllenTech) [reply server/lib/posts.ts]: Damn bot... that'll probably work I just dont know if I like all the repeated code...

[the remark this answers is not in the document]
Comment (JohnAllenTech) [reply server/lib/posts.ts]: You forgot about the select object. If there's a current user you need to add the bookmarked parameter.

The rest looks good

[the remark this answers is not in the document]
Comment (JohnAllenTech) [reply server/lib/posts.ts]: Let me try this and get back to you. It LGTM but need to test

[the remark this answers is not in the document]
Comment (NiallJoeMaher) [reply server/lib/posts.ts]: @JohnAllenTech did this all work? All good to review? 😂

[the remark this answers is not in the document]
Comment (JohnAllenTech) [reply server/lib/posts.ts]: Will check it out later on today. I think the bot generated code will work but havent tested tbh

[the remark this answers is not in the document]
Comment (JohnAllenTech) [reply server/lib/posts.ts]: @coderrabbitai I implemented your suggestions and made a few tweaks to satisfy typescript and eslint but you were nearly there.

**Comment by JohnAllenTech** [comment]
@NiallJoeMaher ready for review now

**Review by NiallJoeMaher — APPROVED** [review]
🌮 !!!

**Pull request merged**